File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    for i, j in enumerate(prices):
        cnt = 0
        for k in range(i+1,len(prices)):
            if prices[k] >=j:
                cnt += 1
            elif prices[k] < j:
                cnt+=1
                break
        answer.append(cnt)
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;
class Solution {
    public int[] solution(int[] prices) {
        int len = prices.length;
        int[] answer = new int[len];
        Stack<Integer> st = new Stack<>();
        st.push(0);

        for(int i=1; i<len; i++){
            if(st.isEmpty()){
                st.push(i);
            }else{
                if(prices[st.peek()] > prices[i]){
                    while(!st.isEmpty() && prices[st.peek()] > prices[i]){
                        int p = st.pop();
                        answer[p] = i-p;
                    }
                    st.push(i);
                }else{
                    st.push(i);
                }
            }
        }

        if(!st.isEmpty()){
            for(int i : st){
                answer[i] = len-1-i;
            }
        }

        return answer;
    }
}